src/index/src/inverted_index/search/fst_apply/intersection_apply.rs

```diff
@@ -32,7 +36,53 @@ pub struct IntersectionFstApplier {
     ranges: Vec<Range>,
 
     /// A list of `Dfa` compiled from regular expression patterns.
-    dfas: Vec<DFA<Vec<u32>>>,
+    dfas: Vec<DfaFstAutomaton>,
+}
+
+#[derive(Debug)]
+struct DfaFstAutomaton(DFA<Vec<u32>>);
+
+impl fst::Automaton for DfaFstAutomaton {
+    type State = StateID;
+
+    #[inline]
+    fn start(&self) -> Self::State {
+        let config = Config::new().anchored(Anchored::No);
```

Comment:
non-blocking: `anchored(Anchored::No)` is the default config
https://docs.rs/regex-automata/latest/regex_automata/util/start/struct.Config.html#method.anchored
```suggestion
        let config = Config::new();
```

Reply:
I'd prefer make it explicit.

Reply:
This is important to our case for what to match. No matter if it's the default, I'd like to make it explicit.